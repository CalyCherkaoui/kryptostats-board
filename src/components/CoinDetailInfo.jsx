/* eslint-disable arrow-body-style */
/* eslint-disable no-console */
import PropTypes from 'prop-types';
import parse from 'html-react-parser';

const CoinDetailInfo = ({
  infoImage,
  infoName,
  infoSymbol,
  infoDescription,
  infoCurrentPrice,
  infoChange24,
  infoVolume,
}) => {
  const description = `<p>${infoDescription}</p>`;
  const parsedDescription = parse(description);
  return (
    <div className="coin_page_info_wrapper">
      <div className="coin_page_about">
        <img src={infoImage} alt={infoName} className="coin_page_about_image" />
        <div className="coin_page_name_symbol">
          <span>{infoName}</span>
          <span>{infoSymbol}</span>
        </div>
        <div className="coin_page_about_description">{parsedDescription}</div>
        <div className="coin_page_data">
          <div className="coin_page_data_current_price">
            {infoCurrentPrice}
          </div>
          <div className="coin_page_data_total_vol">
            {infoVolume}
          </div>
          <div className="coin_page_data_change24h">
            {infoChange24}
          </div>
        </div>
      </div>
    </div>
  );
};

CoinDetailInfo.propTypes = {
  infoName: PropTypes.string.isRequired,
  infoSymbol: PropTypes.string.isRequired,
  infoImage: PropTypes.string.isRequired,
  infoDescription: PropTypes.string.isRequired,
  infoCurrentPrice: PropTypes.number.isRequired,
  infoChange24: PropTypes.number.isRequired,
  infoVolume: PropTypes.number.isRequired,
};

export default CoinDetailInfo;
