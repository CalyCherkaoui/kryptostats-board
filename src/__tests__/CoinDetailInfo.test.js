import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import CoinDetailInfo from '../components/CoinDetailInfo';
import store from '../redux/store';

describe(
  'Renders correctly CoinDetailInfo component',
  () => {
    let container;
    beforeEach(() => {
      container = shallow(
        <Provider store={store}>
          <Router>
            <CoinDetailInfo />
          </Router>
        </Provider>,
      );
    });

    it(
      'Contains Coin Info Header',
      () => {
        expect(container.find('.coin_info_header')).toBeTruthy();
      },
    );

    it(
      'Contains coin info market',
      () => {
        expect(container.find('.coin_info_market')).toBeTruthy();
      },
    );

    it(
      'Contains coin description',
      () => {
        expect(container.find('.coin_description')).toBeTruthy();
      },
    );

    it(
      'Matches CoinDetailInfo snapshot',
      () => {
        const rendered = renderer.create(
          <Provider store={store}>
            <Router>
              <CoinDetailInfo />
            </Router>
          </Provider>,
        );
        expect(rendered).toMatchSnapshot();
      },
    );
  },
);
