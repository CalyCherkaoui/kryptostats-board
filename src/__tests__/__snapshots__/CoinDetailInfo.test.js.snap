// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly CoinDetailInfo component Matches CoinDetailInfo snapshot 1`] = `
<div>
  <p
    className="nodata"
  >
    No Data available for your input...
  </p>
</div>
`;
